ql/src/java/org/apache/hadoop/hive/ql/parse/repl/dump/events/InsertHandler.java

```diff
@@ -81,10 +82,22 @@ public void handle(Context withinContext) throws Exception {
         dataPath = new Path(withinContext.eventRoot, qlPtns.get(0).getName());
       }
 
-      // encoded filename/checksum of files, write into _files
-      try (BufferedWriter fileListWriter = writer(withinContext, dataPath)) {
-        for (String file : files) {
-          fileListWriter.write(file + "\n");
+      int repeat = 0;
+      boolean done = false;
+
+      while (!done) {
+        // encoded filename/checksum of files, write into _files
+        try (BufferedWriter fileListWriter = writer(withinContext, dataPath)) {
+          for (String file : files) {
+            fileListWriter.write(file + "\n");
+          }
+          done = true;
+        } catch (IOException e) {
+          // For any error, start the file write again from starting as we don't know if the file is corrupted or not.
+          LOG.info("File write failed with error {}. Num retry {}", e.getMessage(), repeat++);
+          if (repeat == FileOperations.MAX_IO_ERROR_RETRY) {
+            throw e;
```

Comment:
Add an error log saying it failed after several attempts.

Reply:
info is already there for num retry , so i think one more error log will be redundant

Reply:
If user enable ERROR log level, then info won't be printed but need to see the error message.
We shall print, the files details on error and shall remove from retry info logs. Also, need to add a toString implementation in FileInfo class for logs to print it properly.

Reply:
add error with files list